pkg/state/manager.go

```diff
@@ -96,14 +97,20 @@ func (smgr *stateManager) SyncState(customResource interface{}, infoCatalog Info
 			log.V(consts.LogLevelError).Info("Error while syncing states", "Error:", err)
 			return managerResult, err
 		}
+
 		if !done {
+			statesReady = false
 			log.V(consts.LogLevelInfo).Info("State Group Not ready")
-			return managerResult, nil
+		} else {
+			statesReady = true
+			log.V(consts.LogLevelInfo).Info("Sync Completed successfully for State group", "index", i)
 		}
-		log.V(consts.LogLevelInfo).Info("Sync Completed successfully for State group", "index", i)
 	}
-	// Done Syncing CR
-	managerResult.Status = SyncStateReady
-	log.V(consts.LogLevelInfo).Info("Sync Done for custom resource")
+	if statesReady {
```

Comment:
don't you overwrite the statesReady  in the for loop? so here you are checking according to the last one?